### PR TITLE
03 get /api/categories

### PR DESCRIPTION
**Pull request by munist1010**
created error handling suite, test suite for app, created api and api/categories endpoints, created model, controller and app files

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const { getCategories } = require("./gamesController");
+const {
+	handleCustomErrors,
+	handlePsqlErrors,
+	handleServerErrors,
+} = require("./errorHandlingControllers");
+
+const app = express();
+app.use(express.json());
```
Comment (johnhutcheon) [inline app.js:10]: Please remove this as it's not required yet. 😀

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				.get("/api/categories")
+				.expect(200)
+				.then((response) => {
+					const { body } = response;
+					const categories = body;
+					expect(Array.isArray(categories)).toBe(true);
+					expect(categories[0]).toMatchObject({
+						slug: expect.any(String),
+						description: expect.any(String),
+					});
```
Comment (johnhutcheon) [inline __tests__/app.test.js:20]: This only checks the first element, consider using iteration to check every element

File: app.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+app.get("/api/", (req, res) => {
+	res.status(200).send({ msg: "all good!" }).catch(next);
+});
```
Comment (johnhutcheon) [inline app.js:14]: This catch isn't required here, as it isn't doing anything

File: gamesController.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const { fetchCategories } = require("./gamesModel");
+
+exports.getCategories = (req, res) => {
+	fetchCategories().then((categories) => {
+		res.status(200).send(categories);
+	});
```
Comment (johnhutcheon) [inline gamesController.js:6]: Please add the catch here instead of in the app. 🙂

File: __tests__/app.test.js (hunk 2 above)
Comment (munist1010) [reply __tests__/app.test.js:20]: Done!

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Aaron. Nice work so far but could you just check the comments and amend please. Thanks :)

**Pull request merged**